engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Array.java

```diff
@@ -128,6 +131,54 @@ boolean hasType() {
     return true;
   }
 
+  @ExplodeLoop
```

Comment:
Explode look is unlikely to work as `items` isn't a compile time constant. You'd have to use a `ValueProfile.profile(items.length)` or co., but I am not sure how often one processes arrays of the same size? Maybe vector literals?